addons/mail/static/src/core/common/follower_model.js

```diff
@@ -36,6 +36,7 @@ export class Follower extends Record {
             [this.thread.id],
             [this.partner.id],
         ]);
+        this.thread.followersCount -= 1;
```

Comment:
`message_unsubscribe` (or a controller version of it) should return relevant store data instead